src/routes/home.js

```diff
@@ -37,3 +38,14 @@ module.exports = {
     }
   }
 };
+
+const getUserData = (room, endpoint) => {
+  let roomId = Rooms[room];
+  let endpointId = Rooms[room][endpoint];
+
+  let roomName = roomId.getroomName();
+  let username = roomId.endpoints[endpointId].getUsername();
+  let permissions = roomId.endpoints[endpointId].getPermissions();
+
+  return { roomId, roomName, endpointId, username, permissions };
```

Comment:
Instead of returning an object like this, first store a variable to hold everything in;

const viewData = { roomId, roomName, endpointId, username, permissions };

Then check for permissions, if permissions then get the room pin, and store the pin too (for the mentors view)
```javascript
if (roomId.getPermissions().includes('av')) {
   viewData.pin = roomId.getPin()
}

return viewData
```